### PR TITLE
Add VSCode config for project

### PR DESCRIPTION
**Pull request by caiopiccirillo**
### Checklist

- [x] Confirm that `run-checks.sh` has been executed.

### Related Issues/PRs

No issue

### Changes

This changes suggests the addition of some VSCode extensions, which can help us on development and debug.
To begin with, when a new contributor start a fresh VScode instance on burn, he/she will receive the following message to install the suggested extensions:
image

With LLDB extension we can insert breakpoints and debug our changes, just go to "Run and Debug" tab and select the lib or test related to it:
image

To update the `launch.json` file, we should press F1 (or ctrl+shift+P) and type `LLDB: Generate Launch Configurations from Cargo.toml`:
image

### Testing

No test needed

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+                    "--package=burn-core"
+                ],
+                "filter": {
+                    "name": "burn-core",
+                    "kind": "lib"
+                }
+            },
+            "args": [],
+            "cwd": "${workspaceFolder}"
+        },
```
[the remark this answers is not in the document]
Comment (caiopiccirillo) [reply .vscode/launch.json]: I created a script called `.vscode/generate_launch_config.py`, which will run every time the user opens vscode and update the `launch.json` file. But also this script can be executed manually from root folder as needed.
I also removed the `launch.json` from our git track.

[the remark this answers is not in the document]
Comment (Luni-4) [reply .vscode/launch.json]: @caiopiccirillo 

Is it possible to convert `generate_launch_config.py` in `Rust` language? In this way other developers do not need to install `Python` to generate the config.

**Comment by nathanielsimard** [comment]
@antimora I think you are using vscode, I would appreciate your feedback on this 😁

**Comment by antimora** [comment]
> @antimora I think you are using vscode, I would appreciate your feedback on this 😁

I'll tested out and review it.

**Comment by antimora** [comment]
I've given it a try locally, and here's my assessment.

The setup does have its merits, but our approach might be excessively intrusive and could introduce potential security risks due to the automated execution of scripts.

In my view, we might be better served by simply documenting how to debug Rust code using the LLDB extension, an outstanding tool that I've utilized before. Aside from my thoughts, I've encountered the following issues:

- The script fails to execute on my Mac M1 laptop, indicating that Python is missing. This issue can be fixed by using the shebang line `#!/usr/bin/env python3`. Additionally, the script appears to assume that I am operating from the project root.
- After running the script, I notice that the list is populated with too many items. A screenshot of this can be seen below.

image

These observations might warrant consideration as we proceed with this approach.

**Comment by caiopiccirillo** [comment]
So, I'll create some documentation of how to setup, where can I create it?

**Comment by antimora** [comment]
> So, I'll create some documentation of how to setup, where can I create it?

I am not sure at this point but somewhere we can collect our Burn development knowledge (adding new ops, how to debug, etc). One place I could suggest is in our book's section on development. 

@louisfd , @nathanielsimard where should we store such knowledge? I think it's valuable because I have done something similar in the past.

**Comment by nathanielsimard** [comment]
@antimora @caiopiccirillo I think we should document it here: https://github.com/burn-rs/burn/blob/main/CONTRIBUTING.md

**Comment by nathanielsimard** [comment]
I'm closing the PR since the vscode setup should be documented in the CONTRIBUTING file instead of commiting the configs in the repo.

**Pull request closed without merging**